### ISSUE TITLE
Linking to platform guides from the homepage

### ISSUE THREAD

**Issue by robinrodricks** [HaxeFoundation/haxe.org#376]
Hello Haxe community!

As some of you know we now have per-platform installation guides for (somewhat) every platform Haxe supports.

- [C#](https://staging.haxe.org/documentation/platforms/csharp.html)
- [PHP](https://staging.haxe.org/documentation/platforms/php.html )
- [Java](https://staging.haxe.org/documentation/platforms/java.html)
- [Python](https://staging.haxe.org/documentation/platforms/python.html)

How can we link to these from the homepage?

Would it be fine if I made the respective platform logos on the homepage clickable and link to it from there?

**Comment by markknol** [comment]
I would prefer to have all targets in there too before linking them (JavaScript / HashLink / C++ …).

Then we should incorporate it here somewhere: https://haxe.org/documentation/introduction/

**Comment by uvtc** [comment]
Looks like I commented on the PR #377 rather than this issue, but I suggest creating a small install page on haxe.org containing some short instructions regarding grabbing the package for your OS from the Downloads page and installing in the usual manner for that OS, but *also* containing a matrix or list of links to these cool setup guides.

I think a prospective new Haxe user looking to install and setup Haxe will look in the following places, in the following order, to figure out what to do:

  * an Install link in the main nav heading?
  * an Install link under the "Learn Haxe" tab?
  * an Install link in the Introduction?
  * an Install link in the Manual somewhere (maybe)?

@markknol writes: "I would prefer to have all targets in there too before linking them (JavaScript / HashLink / LUA, C++ …)."

That would be nice, but may be a long way out. Maybe acceptable to put a blurb underneath saying, "for targets/platforms not listed here, contributions welcome" or similar.

**Comment by uvtc** [comment]
@markknol Looking at [the Haxe intro](https://haxe.org/documentation/introduction/) now, I'm surprised there's not already a link to "installing" there (above "Building from source"); I think a new user reading the intro is very likely to also want info on installing and setup.

**Comment by robinrodricks** [comment]
@uvtc 
Assuming this workflow:

1. Someone is interested and gets eventually carried off to the haxe download page 
2. He downloads a specific version of haxe corresponding to his platform
3. He gets taken to the "thank you for downloading haxe" page
4. At this point he could presumably use some platform-specific setup instructions

Thus I modified my PR to include some changes to the "thanks for download" page that look like:

![image](https://user-images.githubusercontent.com/6797866/55132358-244c8300-5148-11e9-8408-d8f5c369292b.png)

Does it work?

**Comment by Simn** [comment]
I really like the idea to have this on the download page. Seems like the perfect place for it!

Could we do just that without changing anything on the index page for now?

**Comment by robinrodricks** [comment]
Could we have both? :P

**Comment by robinrodricks** [comment]
@markknol Shall I revert the index page changes?

**Comment by markknol** [comment]
Yes, however please don't remove the branch, I might want to use the separate logos when updating the homepage one day.

**Comment by robinrodricks** [comment]
Lol thanks! I commented out the platform logos from the homepage while keeping the source in-place for future usage. The homepage is now visually untouched.

And I removed the non-real target logos.

**Comment by markknol** [comment]
Will look at it next week as I'm little bit busy atm. I first want to see the buttons, because once it is on stagin we cannot deploy other things (if needed) without shipping this along to live website, if that makes sense.

**Comment by robinrodricks** [comment]
We also need some link to these guides from within the main docs menu / page.

**Comment by robinrodricks** [comment]
Done the buttons. Hopefully you gonna accept it now?

**/documentation/introduction/**

![image](https://user-images.githubusercontent.com/6797866/55271927-fe0b1c80-52da-11e9-9abd-c488cbb135c8.png)

**Post download page**

![image](https://user-images.githubusercontent.com/6797866/55271931-0b280b80-52db-11e9-89d8-3efd161d4758.png)

Are we good?

**Comment by robinrodricks** [comment]
@markknol Is dis fine?

**Comment by markknol** [comment]
This is great! Looks very good, nice job! 🔥 

Now I still think we need JavaScript and C++ since those are important targets.

**Comment by markknol** [comment]
Another thing that we might want to update on the pages is the "Ask the community" section, there is more than the forum, there is stackoverflow, discord, gitter, even social media https://haxe.org/foundation/contact.html

I hope we can this live soon !

**Comment by robinrodricks** [comment]
Good ideas, however I'd like to keep it above the fold, you know, people may not scroll below the text and may miss those buttons. I might move it just below the donate/technical help sections.

**Comment by markknol** [comment]
I doubt  that argument about the fold is an actual thing, people scroll, also depends on a resolution if it will be visible anyway. I think it belongs to the other "getting started" section, so lets incorporate it there.

**Comment by robinrodricks** [comment]
Great. Now we've shoved these buttons from the homepage down to the end of a minor download page. I hope people find it after all this work.

**Comment by markknol** [comment]
> Anyways, its your site, so whatever...

That's a bit dramatic, and there is no need for that 😃 You're doing great work and we both want to make it as visible as possible, on the correct spots. I'm open for suggestions. 
Also I though we wanted those buttons here too? https://haxe.org/documentation/introduction/ Which is the 2nd most visited page on the haxe.org site.

**Comment by robinrodricks** [comment]
Haha. Thanks. Yeah look, I just hope people find it below the fold. Yeah most people may be using a large HD screen and they could possibly find those buttons. Its just personally I would filter out any and all text on the "thanks for downloading" page especially after I see those big "donate" and "contact us" sections. I would just assume that anything coming after that could not help me majorly. 

And yeah, I appreciate your help too bud. I already included the changes to **/documentation/introduction/** in my post above, let me recap:

![](https://user-images.githubusercontent.com/6797866/55271927-fe0b1c80-52da-11e9-9abd-c488cbb135c8.png)

Is this good? And I'll try to do what you're saying... lets see...

**Comment by robinrodricks** [comment]
How is this for the post download page?

![test](https://user-images.githubusercontent.com/6797866/55556842-22655f80-5706-11e9-8fc6-f9f278286818.gif)

**Comment by markknol** [comment]
That looks good! Glad we're on the same track!

**Comment by robinrodricks** [comment]
Ok, great! Now just to complete our selection of pages. Flash and JS should be easy. C++ might need some work.

**Comment by markknol** [comment]
I added the JS + node.js page https://github.com/HaxeFoundation/haxe.org/commit/d191fe4fc253f75b49d29da25abbfb986d081c13 and changed the link to the community support https://github.com/HaxeFoundation/haxe.org/commit/76da31082b9426eead693ac5597189b837809a46
Can you check them out?

**Comment by markknol** [comment]
Btw it would be great if we could re-use the SVG icons (and color them with css) on the platform pages.

**Comment by robinrodricks** [comment]
> I added the JS + node.js page [d191fe4](https://github.com/HaxeFoundation/haxe.org/commit/d191fe4fc253f75b49d29da25abbfb986d081c13) and changed the link to the community support [76da310](https://github.com/HaxeFoundation/haxe.org/commit/76da31082b9426eead693ac5597189b837809a46)
> Can you check them out?

Thanks so much. It was getting tiring to create those pages. I think they look great. The node.js page may need a section on installing node.js, like we have in java/python.

**Comment by markknol** [comment]
On my local machine I added a feature to include a piece of html so you can do  `::fragment "pages/community.fragment.html"::`

This allows me to include this, without repeating this on each page, what do you think? 

![image](https://user-images.githubusercontent.com/576184/55563132-a105d100-56f5-11e9-8d0d-0beda0653946.png)

**Comment by robinrodricks** [comment]
Its nice but I probably think its overkill. I mean when faced 20 links, I would click none. Plus, many of the links are overlapping with the API section above. I would rather not have this template, or a very cut down version of it with just the actually useful ones. For example - Compiler bug reports - Unlikely that newbies need this. Blog/Facebook/Twitter is once again useless. The docs are already covered above.

**Comment by robinrodricks** [comment]
I'm kinda done with this PR. I improved the button style to more closely match the ones on the rest of the site. I added JS and Node.js.

Could you accept it into staging?

**Documentation page**

![image](https://user-images.githubusercontent.com/6797866/55700422-cdb73280-59ec-11e9-87ab-83c93cc02b28.png)

**Post download page**

![image](https://user-images.githubusercontent.com/6797866/55700412-c728bb00-59ec-11e9-93f2-a458f5eac1f7.png)

**Comment by markknol** [comment]
will review when I have some time! Thanks!
